src/components/Dropdown/Dropdown.test.tsx

```diff
@@ -0,0 +1,523 @@
+          const onChangeMock = jest.fn();
+          const user = userEvent.setup();
+          const selectedOption = defaultProps.options[1]!;
+          render(<Dropdown {...defaultProps} onChange={onChangeMock} />);
+          expect(onChangeMock).not.toHaveBeenCalled();
+
+          await user.click(screen.getByRole('combobox'));
+          await user.keyboard(`{${KeyCodes.ARROW_DOWN}}`);
+          await user.keyboard(SPACE_SYMBOL);
+          expect(onChangeMock).toHaveBeenCalledWith(
+            selectedOption.id,
+            selectedOption.value,
+          );
+        });
+      });
+
+      it('should close the option list when an option is selected', async () => {
+        const user = userEvent.setup();
+        const selectedOption = defaultProps.options[1]!;
+        render(<Dropdown {...defaultProps} />);
+
+        await user.click(screen.getByRole('combobox'));
+        expect(screen.getByRole('combobox')).toHaveAttribute(
+          'aria-expanded',
+          'true',
+        );
+
+        await user.click(
+          screen.getByText(selectedOption.value, {
+            selector: '[role=option]',
+          }),
+        );
+        expect(screen.getByRole('combobox')).toHaveAttribute(
+          'aria-expanded',
+          'false',
+        );
+      });
+    });
+  });
+});
+
+describe('onKeyDown', () => {
+  const defaultParams: OnKeyDownParams = {
+    event: {
+      preventDefault: jest.fn(),
+      code: '',
+    } as unknown as KeyboardEvent<HTMLDivElement>,
+    options: defaultProps.options,
+    isMenuOpen: false,
+    setIsMenuOpen: jest.fn(),
+    focusedOptionIndex: 0,
+    setFocusedOptionIndex: jest.fn(),
+    onChange: jest.fn(),
+  };
+
+  describe(KeyCodes.ARROW_UP, () => {
+    const code = KeyCodes.ARROW_UP;
+
+    it('should call preventDefault() on the provided event', () => {
+      expect(defaultParams.event.preventDefault).not.toHaveBeenCalled();
+      onKeyDown({
+        ...defaultParams,
+        event: { ...defaultParams.event, code },
+      });
+      expect(defaultParams.event.preventDefault).toHaveBeenCalled();
+    });
+
+    it('should call the provided setFocusedOptionIndex() with the provided focusedOptionIndex - 1 if focusedOptionIndex!=0', () => {
+      const focusedOptionIndex = 2;
+      expect(defaultParams.setFocusedOptionIndex).not.toHaveBeenCalled();
+      onKeyDown({
+        ...defaultParams,
+        event: { ...defaultParams.event, code },
+        focusedOptionIndex,
+      });
+      expect(defaultParams.setFocusedOptionIndex).toHaveBeenCalledWith(
+        focusedOptionIndex - 1,
+      );
+    });
+
+    it('should not call the provided setFocusedOptionIndex() if the provided focusedOptionIndex=0', () => {
+      onKeyDown({
+        ...defaultParams,
+        event: { ...defaultParams.event, code },
+        focusedOptionIndex: 0,
+      });
```

Comment:
Does this mean the active option is the first element of the array or that no option is selected?

Reply:
This means that the first element of the array is the active option